railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -515,13 +519,13 @@ def app_const
 
       def valid_const?
         if app_const =~ /^\d/
-          raise Error, "Invalid application name #{app_name}. Please give a name which does not start with numbers."
-        elsif RESERVED_NAMES.include?(app_name)
-          raise Error, "Invalid application name #{app_name}. Please give a " \
+          raise Error, "Invalid application name #{original_app_name}. Please give a name which does not start with numbers."
```

Comment:
> Kept original app name which use exception message.

Is there any reason don't normalize `0-foo` to `0_foo` unlike `0.foo`?

Reply:
Because this message display on command line (When it execute `rails new` with invalid args), so user doesn't know process why his/her arguments is normalized.